Update Developer Guide

shuyangk opened the pull request:
The current Developer Guide is modelled after AB3, which does not match the project at hand.

There is a need to change it to suit the current project and contain the relevant information.

Let's update the Developer Guide as necessary to change the required details.

shuyangk committed:
Update Developer Guide

The current Developer Guide is modelled after AB3, which does not
match the project at hand.

There is a need to change it to suit the current project and contain
the relevant information.

Let's update the Developer Guide as necessary to change the required
details.

leawang0 commented on docs/DeveloperGuide.md:
```diff
@@ -257,42 +257,51 @@ _{Explain here how the data archiving feature will be implemented}_
 ### Use cases
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+(For all use cases below, the **System** is `BandConnect+` and the **Actor** is the `producer`, unless specified otherwise)
```
Is our app BandConnect++ with two pluses hahaha

leawang0 commented on docs/DeveloperGuide.md:
```diff
@@ -304,7 +313,30 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+
+* 3a. BandConnect+ detects an error in the entered data.
+
+    * 3a1. BandConnect+ shows an error message.
```
lgtm!

leawang0 reviewed (changes requested):
Only the nit on "++" loll, otherwise lgtm!

shuyangk committed:
Edit product name in developer guide

Product name was written wrongly in earlier iteration of developer
guide.

Product name changed from BandConnect+ to BandConnect++

tanyyyming commented on docs/DeveloperGuide.md:
```diff
@@ -257,42 +257,51 @@ _{Explain here how the data archiving feature will be implemented}_
 ### Use cases
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+(For all use cases below, the **System** is `BandConnect++` and the **Actor** is the `producer`, unless specified otherwise)
```
The actor is the "music producer" to be more accurate. This can be modified in a later version.

tanyyyming commented on docs/DeveloperGuide.md:
```diff
@@ -257,42 +257,51 @@ _{Explain here how the data archiving feature will be implemented}_
+| `*`      | producer who has worked with a musician before   | give a musician a rating                         | can keep track of my past experience with them                         |
+| `*`      | producer who is actively producing music         | check the availability of a musician             | I can avoid musicians with clashing schedules                          |
+| `*`      | producer with many musicians in the address book | sort persons by name                             | locate a person easily                                                 |
+
```
More user stories to be added. I will add some according to our excel sheet after this skeleton is merged.

tanyyyming reviewed (approved):
There are some places to be edited in a future version, but they are just small changes. Otherwise, LGTM!

yytan25 commented on docs/DeveloperGuide.md:
```diff
@@ -321,6 +353,7 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 * **Mainstream OS**: Windows, Linux, Unix, OS-X
+* **Genre**: Genre of music, including but not limited to: Jazz, Classical, Rock
 * **Private contact detail**: A contact detail that is not meant to be shared with others
```
As we add more features to the product, we can add on to this list as well!

yytan25 reviewed (approved):
LGTM!
